src/integration-test/java/com/commercetools/sync/integration/commons/utils/ProductITUtils.java

```diff
@@ -113,33 +120,63 @@ public static void deleteAllProducts(@Nonnull final SphereClient ctpClient) {
+     * Gets the supplied {@link ProductDraft} with the price reference attached on all its variants' prices.
      *
      * @param productDraft     the product draft to attach the channel reference on its variants' prices.
      * @param channelReference the channel reference to attach on the product draft's variants' prices.
-     * @return the product draft with the supplied channel reference attached on the product draft's variants' prices.
+     * @return the product draft with the supplied references attached on the product draft's variants' prices.
      */
-    public static ProductDraft getDraftWithPriceChannelReferences(@Nonnull final ProductDraft productDraft,
-                                                                  @Nonnull final Reference<Channel> channelReference) {
+    public static ProductDraft getDraftWithPriceReferences(@Nonnull final ProductDraft productDraft,
+                                                           @Nullable final Reference<Channel> channelReference,
+                                                           @Nullable final CustomFieldsDraft customFieldsDraft) {
         final List<ProductVariantDraft> allVariants = productDraft
             .getVariants().stream().map(productVariant -> {
-                final List<PriceDraft> priceDraftsWithChannelReferences =
-                    productVariant.getPrices().stream()
-                                  .map(price -> PriceDraftBuilder.of(price).channel(channelReference).build())
-                                  .collect(toList());
+                final List<PriceDraft> priceDraftsWithChannelReferences = productVariant
+                    .getPrices()
+                    .stream()
+                    .map(PriceDraftBuilder::of)
+                    .map(priceDraftBuilder -> ofNullable(channelReference).map(priceDraftBuilder::channel)
+                                                                          .orElse(priceDraftBuilder))
+                    .map(priceDraftBuilder -> ofNullable(customFieldsDraft).map(priceDraftBuilder::custom)
+                                                                           .orElse(priceDraftBuilder))
+                    .map(PriceDraftBuilder::build)
+                    .collect(toList());
                 return ProductVariantDraftBuilder.of(productVariant)
                                                  .prices(priceDraftsWithChannelReferences)
                                                  .build();
             })
             .collect(toList());
         final List<PriceDraft> masterVariantPriceDrafts = productDraft
-            .getMasterVariant().getPrices().stream().map(price -> PriceDraftBuilder.of(price)
-                                                                                   .channel(channelReference)
-                                                                                   .build()).collect(toList());
+            .getMasterVariant().getPrices().stream()
```

Comment:
Maybe this common part you can put it in a method:

```java
productVariant.getPrices().stream()
            .map(PriceDraftBuilder::of)
            .map(priceDraftBuilder -> ofNullable(channelReference).map(priceDraftBuilder::channel)
                                                                  .orElse(priceDraftBuilder))
            .map(priceDraftBuilder -> ofNullable(customFieldsDraft).map(priceDraftBuilder::custom)
                                                                   .orElse(priceDraftBuilder))
            .map(PriceDraftBuilder::build)
            .collect(toList())
```

Reply:
6619e70